bigquery/client.py

```diff
@@ -276,7 +276,7 @@ def _insert_job(self, body_object):
             body=body_object
         ).execute()
 
-    def query(self, query, max_results=None, timeout=0, dry_run=False, use_legacy_sql=None):
+    def query(self, query, max_results=None, timeout=0, dry_run=False, use_legacy_sql=None, external_udf_uris=[]):
```

Comment:
We shouldn't use lists as a default value for kwargs (see http://reinout.vanrees.org/weblog/2012/04/18/default-parameters.html). This should default to `None` and check if it's `None` and assign it to an empty list.